### DISCUSSION TITLE
Vite React Code Hiding

### DISCUSSION THREAD

**Discussion by serhatzunluoglu** [vitejs/vite-plugin-react#338] [Q&A] [answered]
In my **react** project that I created with **Vite**, I do not want the codes to appear in the sources section in the browser. How can I do that? I tried things like sourcemap=false but it didn't work.
![Ekran görüntüsü 2024-06-19 184316](https://github.com/vitejs/vite-plugin-react/assets/97339604/d94d0679-1c2d-481b-8697-205da8efddfa)

**Comment by hi-ogawa** [comment]
This is likely a limitation of Vite's non-bundled approach https://github.com/vitejs/vite/discussions/16440#discussioncomment-9136161
You see two where one is the transpiled file served by Vite and the other one is the original source shown by browser via sourcemap.

**Reply by serhatzunluoglu** [reply]
So how can I solve this problem?

**Reply by Shakeskeyboarde** [reply]
Try building and then starting a preview server instead of running the vite dev server.

What @hi-ogawa is saying is that dev mode always includes source maps. Vite's dev and build modes are only loosely related, and `build` configuration is not fully applied in the dev server. Setting `build.sourcemaps` to false should disable them in _production_ code, which you can verify by building and then starting the preview server.

**Comment by ArnaudBarre** [comment, marked as answer]
I don't understand, what is the problem of having sources present in development?
This will not be present in the prod version if you disable source maps (or push them to something like Sentry and then remove them from the dist folder)
